.github/workflows/inspektor-gadget.yml

```diff
@@ -78,100 +127,207 @@ jobs:
+    - name: Cache debian packages
+      uses: actions/cache@v2
+      id: cache-debs
+      with:
+        path: "~/cache-debs"
+        key: cache-debs-libbpf-1:0.4.0-1ubuntu1-libseccomp-2.5.1-1ubuntu1~20.04.1
+    - name: Install debian packages
+      # ALERT This action must be run after code was checkout otherwise it will
+      # not find this file.
+      uses: ./.github/actions/install-debian-packages
+    - name: Build ${{ matrix.local-gadget-target }}
       run: |
-        make manifests generate generate-documentation
-        git diff --exit-code HEAD --
+        make ${{ matrix.local-gadget-target }}
 
+        # Prepare assets for release and actions artifacts
+        platform=$(echo ${{ matrix.local-gadget-target }} | cut -d- -f3-4)
+        mkdir $platform
+        cp ${{ matrix.local-gadget-target }} $platform/local-gadget
+        cp LICENSE $platform/
+        tar --sort=name --owner=root:0 --group=root:0 \
+          -czf ${{ matrix.local-gadget-target }}.tar.gz -C $platform \
+          local-gadget LICENSE
+        rm -rf $platform
+    - name: Add ${{ matrix.local-gadget-target }}.tar.gz as artifact.
+      uses: actions/upload-artifact@master
+      with:
+        name: ${{ matrix.local-gadget-target }}-tar-gz
+        path: /home/runner/work/inspektor-gadget/inspektor-gadget/${{ matrix.local-gadget-target }}.tar.gz
+
+  build-docker-image:
+    name: Build gadget container image
+    # level: 0
+    runs-on: ubuntu-latest
+    steps:
     - name: Set up Docker Buildx
       id: buildx
       uses: docker/setup-buildx-action@v1
-
     - name: Cache Docker layers
       uses: actions/cache@v2
       with:
         path: /tmp/.buildx-cache
         key: ${{ runner.os }}-single-buildx-${{ hashFiles('gadget.Dockerfile') }}
         restore-keys: |
           ${{ runner.os }}-single-buildx
-
     - name: Login to Container Registry
       uses: docker/login-action@v1
       with:
         registry: ${{ secrets.CONTAINER_REGISTRY }}
         username: ${{ secrets.CONTAINER_REGISTRY_USERNAME }}
         password: ${{ secrets.CONTAINER_REGISTRY_PASSWORD }}
-
+    - name: Check out code
+      uses: actions/checkout@v1
+    - name: Set IMAGE_TAG
+      run: |
+        TMP1=${GITHUB_REF#*/}
+        TMP2=${TMP1#*/}
+        IMAGE_TAG=${TMP2//\//-}
+        if [ "$IMAGE_TAG" = "main" ]; then
+            IMAGE_TAG="latest"
+        fi
+        echo IMAGE_TAG=$IMAGE_TAG >> $GITHUB_ENV
     - name: Build gadget container
       uses: docker/build-push-action@v2
       with:
-        context: .
-        file: gadget.Dockerfile
+        context: /home/runner/work/inspektor-gadget/inspektor-gadget/
+        file: /home/runner/work/inspektor-gadget/inspektor-gadget/gadget.Dockerfile
         build-args: |
           ENABLE_BTFGEN=true
         # TODO: how to avoid pushing a container before running integration tests
+        # Answer: push to runner registry first, if integration tests are OK
+        # push to final registry
```

Comment:
I am not sure that the `test-integration` job would have access to a registry that is in the `build-docker-image` job runner. Is it what you mean? Could you please explain this idea a little bit more? Thanks!